### ISSUE TITLE
Latest kubernetes provider breaks the auth.tf

### ISSUE THREAD

**Issue by qlikcoe** [terraform-google-modules/terraform-google-kubernetes-engine#429]
`kubernetes` provider v1.11 was released 3 hours ago. Right away it breaks kubernetes provider configuration in `auth.tf`: 

```
Error: Failed to initialize config: invalid configuration: no configuration has been provided
  on .terraform/modules/gke/terraform-google-modules-terraform-google-kubernetes-engine-064f308/modules/private-cluster/auth.tf line 29, in provider "kubernetes":
  29: provider "kubernetes" {
```

Pinning it to v1.10 is a current workaround: 

```
provider "kubernetes" {
  version = "= 1.10"
}
```

**Comment by morgante** [comment]
Upstream issue: https://github.com/terraform-providers/terraform-provider-kubernetes/issues/759

**Comment by lawliet89** [comment]
If you want to keep using 1.11 provider, you can do a partial plan first with something like `terraform plan -target module.gke.google_container_cluster.primary`.

**Comment by EricHorst** [comment]
Upstream issue has been resolved with 1.11.1: terraform-providers/terraform-provider-kubernetes#759

@morgante Would you consider moving the kubernetes provider version up to ">= 1.11.1"?

In my shop we are locked at kubernetes provider 1.9 due to terraform-providers/terraform-provider-kubernetes/issues/787 which was just fixed in 1.11.1. This means we're also stuck on terraform-google-kubernetes-engine 6.2 since 7.x is locked on kubernetes provider 1.10. Moving up to 1.11.1 should get beyond both problems. We desperately want some of the 7.3 knobs.

Thanks.

**Comment by morgante** [comment]
I'd be happy to accept/review a PR which increases/changes the provider version constraint assuming tests pass.
